frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,275 @@ export class OutputArtifactLoader {
     };
   }
 
+  public static async getMlmdContext(argoPodName: string): Promise<Context> {
+    if (argoPodName.split('-').length < 3) {
+      throw new Error('kfpPodName has fewer than 3 parts');
+    }
+
+    // argoPodName has the general form "pipelineName-workflowId-executionId".
+    // All components of a pipeline within a single run will have the same
+    // "pipelineName-workflowId" prefix.
+    const pipelineName = argoPodName
+      .split('-')
+      .slice(0, -2)
+      .join('_');
+    const runID = argoPodName
+      .split('-')
+      .slice(0, -1)
+      .join('-');
+    const contextName = pipelineName + '.' + runID;
+
+    const request = new GetContextByTypeAndNameRequest();
+    request.setTypeName('run');
+    request.setContextName(contextName);
+    let res: GetContextByTypeAndNameResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getContextByTypeAndName(request);
+    } catch (err) {
+      err.message = 'Failed to getContextsByTypeAndName: ' + err.message;
+      throw err;
+    }
+
+    const context = res && res.getContext();
+    if (!context) {
+      throw new Error("getContextByTypeAndName didn't have a context");
```

Comment:
I think you mentioned MLMD doesn't have corresponding data is expected error, so we should simply return context here. (You need to change return type to Promise<Context|undefined>.)